### PR TITLE
get_closest fix

### PR DESCRIPTION
**Pull request by jonyfries**
the ai_tools.get_closest function did not correctly return the closest
item and took a long time to do.

### CODE REVIEW

File: core/player.py [hunk 1]
```diff
@@ -62,6 +62,7 @@ def __init__(self, bullet_group, effects_group, life_board, score_board):
         self.hitrect_offset_y = -15
         self.score = 0
         self.lives = 3
+        #TODO: rework how player lives are set as magic numbers are used here and in the main game logic.
```
Comment (joshuaskelly) [inline core/player.py]: Is this related to the bug fix?

Comment (jonyfries) [reply core/player.py]: Sorry no, note to self for later. Should not be part of this branch.

Comment (jonyfries) [inline core/player.py]: Sorry no. That was a note to self for later.

Comment (joshuaskelly) [reply core/player.py]: Click the issues tab and create an issue if it is something you'd like to fix. Github uses issues as both bugs and feature requests.

Comment (jonyfries) [inline core/player.py]: Sorry no, that was a note for later and shouldn't be include in this.

File: utils/aitools.py [hunk 2]
```diff
@@ -130,89 +130,36 @@ def cardinal_direction(actor):
...
+    :param actors_type_list: The actor types you want to test, expects a list of actor types
+    :return: The closest actor with type in actors_type in test_groups
+    """
+    test_actor_list = []
+
+    if not actor_type_list:
+        for test_group in test_group_list:
+            for test_actor in test_group:
+                test_actor_list.append(test_actor)
+    else:
```
Comment (joshuaskelly) [inline utils/aitools.py:149]: Newline before else clause.

File: utils/aitools.py [hunk 3]
```diff
@@ -130,89 +130,36 @@ def cardinal_direction(actor):
...
-
-            except:
-                if target.actor_type == target_actors:
-                    temp_group2.append(target)
-
-        else:
-            for target in target_group:
-                temp_group2.append(target)
+def get_closest(search_actor, test_group_list, actor_type_list=None):
+    """
```
Comment (joshuaskelly) [inline utils/aitools.py]: Please clean up this docstring. Example of what I'm looking for:
https://gist.github.com/JoshuaSkelly/6d5e6d9f65c7c5dbae90d0360c306aa0

File: utils/aitools.py [hunk 4]
```diff
@@ -131,11 +131,12 @@ def cardinal_direction(actor):
 
 # AI tools
 def get_closest(search_actor, test_group_list, actor_type_list=None):
-    """
-    :param search_actor: The actor that you want to find the closest other actor
-    :param test_group_list: The actor groups that you want to test, expects a list of lists
-    :param actors_type_list: The actor types you want to test, expects a list of actor types
-    :return: The closest actor with type in actors_type in test_groups
+    """Returns closest actor or self
```
Comment (joshuaskelly) [inline utils/aitools.py]: This is supposed to be the description of the method, not the return value.

File: utils/aitools.py [hunk 5]
```diff
@@ -131,11 +131,12 @@ def cardinal_direction(actor):
...
+    
+    Arguments:
+    search_actor: The actor that you want to find the closest other actor
+    test_group_list: The actor groups that you want to test, expects a list of lists
+    actors_type_list: The actor types you want to test, expects a list of actor types
     """
```
Comment (joshuaskelly) [inline utils/aitools.py]: Missing Returns documentation

**Review by joshuaskelly — COMMENTED** [review]
Can you see them now?

**Comment by jonyfries** [comment]
Found it!
